src/util.rs

```diff
@@ -44,7 +44,7 @@ impl<C> Service<Destination> for Connector<C>
 where
     C: Connect,
 {
-    type Response = C::Transport;
+    type Response = (C::Transport, ());
     type Error = C::Error;
     type Future = ConnectorFuture<C>;
```

Comment:
This should be changeable to `(C::Transport, C::Extra)` https://docs.rs/hyper/0.12.25/hyper/client/connect/trait.Connect.html#associatedtype.Error

Then at this point this service can just wrap the future `C::Future` since it should match 1-1. Then all you need to do is `impl HttpExtra for Connected`.